core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -1012,6 +1018,9 @@ protected synchronized Double convertStringToNumber( String string ) throws Kett
 
     try {
       DecimalFormat format = getDecimalFormat( false );
+      if ( string.indexOf( format.getDecimalFormatSymbols().getDecimalSeparator() ) > 0  ) {
```

Comment:
As I understand the "maximum faction digits" setting, it's unnecessary to be so precise and set it to the length of the fractional part. The better (and less costly performance-wise) is to set it to 100 or so. Then no precision is lost, and you don't have to strip off any zeros. See my comment on this case: https://jira.pentaho.com/browse/ANALYZER-3728?focusedCommentId=338607&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-338607